csharp/src/Microsoft.ML.OnnxRuntime/InferenceSession.shared.cs

```diff
@@ -773,15 +773,23 @@ public ulong ProfilingStartTimeNs
 
             if (prepackedWeightsContainer == null)
             {
-                NativeApiStatus.VerifySuccess(NativeMethods.OrtCreateSession(envHandle, NativeMethods.GetPlatformSerializedString(modelPath),
+                var modelPathBytes = NativeOnnxValueHelper.GetPlatformSerializedString(modelPath);
+                using (PinnedGCHandle pinnedTrainHandle = new PinnedGCHandle(GCHandle.Alloc(modelPathBytes, GCHandleType.Pinned)))
+                {
+                    NativeApiStatus.VerifySuccess(NativeMethods.OrtCreateSession(envHandle, modelPathBytes,
```

Comment:
>modelPathBytes [](http://example.com/codeflow?start=92&length=14)

I think something is off. The API should accept IntPtr, which should be obtained from the pinned handle.

Reply:
reverting as discussed offline